samza-yarn/src/main/scala/org/apache/samza/job/yarn/ClientHelper.scala

```diff
@@ -350,6 +350,11 @@ class ClientHelper(conf: Configuration) extends Logging {
       } else {
         false
       }
+    } catch {
+      case e: IOException => {
+        warn("Exception when querying AM metrics. Likely because YARN was re-starting a new AM attempt", e)
```

Comment:
an IOException in querying AM could be because of a few other reasons as well, right?

Reply:
> an IOException in querying AM could be because of a few other reasons as well, right?

I've modified the log line.

> maybe we want to catch any exception?

 probably not - for eg: parse errors due to an invalid format in the response should be bubbled-up